assets/js/icons/icon/index.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * External dependencies
+ */
+import { cloneElement } from '@wordpress/element';
+
+function Icon( { srcElement, size = 24, ...props } ) {
+	return cloneElement( srcElement, {
+		width: size,
+		height: size,
+		...props,
+	} );
```

Comment:
I wonder if we should check that `srcElement` is a valid react element before using (you can use `isElement` from the `@wordpress/element` package for this).